basics.js

```diff
@@ -3,33 +3,42 @@
 // prefixed with the text "I have a " (so "I have a cat", "I have a dog", etc.)
 
 var pets = ['cat', 'dog', 'wombat'];
+function animalArray(){
+  for (var i=0; i < pets.length; i++){
+    console.log("I have a " + pets[i]);
+  }
+};
+animalArray();
 
 ///// Question 2
 // Change my friend's hair color to "green", without modifying the line below.
 
 var friend = { name: 'Chris', age: 27, hairColor: 'blue' };
+friend.haircolor = null;
+console.log(friend);
 
 ///// Question 3
 // The function below is problematic. In 1-2 sentences explain why, and then fix the code.
 
-var squarePlusOne = function(n){
-  squared = n * n;
-  return squared + 1;
+//  'squared' is being built as a global variable, which is bad practice
+
+function squarePlusOne(n){
+  var f = (n * n) + 1;
+  return f;
 };
 
 ///// Question 4
 // Translate the Ruby method defined below into a JavaScript function.
 
-/*
-def greet(name, excited)
-  greeting = "Hi, #{name} here!"
-  if excited == true
-    greeting.upcase
-  else
-    greeting
-  end
-end
-*/
+
+function greet(name, excited){
+  var greeting = "Hi, " + name + " here!"
+  if excited === true {
+      greeting.uppercase;
```

Comment:
You're looking for the .toUpperCase function, and you'll need to actually invoke it with () at the end.